dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractHostedEntryResource.java

```diff
@@ -120,7 +122,12 @@ public T editHosted(@ApiParam(hidden = true) @Auth User user,
         checkUser(user, entry);
         checkHosted(entry);
         U version = getVersion(entry);
-        handleSourceFileMerger(entryId, sourceFiles, entry, version);
+        Set<SourceFile> versionSourceFiles = handleSourceFileMerger(entryId, sourceFiles, entry, version);
+        boolean isValidVersion = checkValidVersion(versionSourceFiles, entry);
+        if (!isValidVersion) {
+            throw new WebApplicationException("The reversion is not valid", HttpStatus.SC_BAD_REQUEST);
```

Comment:
I think this makes it into the GUI, so maybe something more descriptive like 
"Your edited files for are invalid. No new version was created.  Please check your syntax and try again."